src/Microsoft.OData.Client/BaseSaveResult.cs

```diff
@@ -495,7 +495,7 @@ protected override void AsyncEndGetResponse(IAsyncResult asyncResult)
                 {
                     this.HandleOperationResponse(responseMessage);
 
-                    if (!Util.IsBulkUpdate(this.Options))
+                    if (!Util.IsBulkUpdate(this.Options) && !Util.IsDeepInsert(this.Options))

```

Comment:
Should this be `&&` or `||`.

Reply:
`&&` since the condition is checking when we don't have `Bulk update` and `Deep insert`

Reply:
so what happens when we have either? Will this return false? I think we also need a test that has both a bulk update and a bulk insert.

Reply:
`True` and `False` is `False` so that code won't be hit.